pkg/trace/watchdog/cpu_aix.go

```diff
@@ -0,0 +1,107 @@
+//             pid64_t  pr_ppid;                   /* parent process id */
+//             pid64_t  pr_pgid;                   /* process group id */
+//             pid64_t  pr_sid;                    /* session id */
+//             struct pr_timestruc64_t pr_utime;   /* process user cpu time */
+//             struct pr_timestruc64_t pr_stime;   /* process system cpu time */
+//             struct pr_timestruc64_t pr_cutime;  /* sum of children's user times */
+//             struct pr_timestruc64_t pr_cstime;  /* sum of children's system times */
+//             pr_sigset_t pr_sigtrace;            /* mask of traced signals */
+//             fltset_t pr_flttrace;               /* mask of traced hardware faults */
+//             uint32_t pr_sysentry_offset;        /* offset into pstatus file of sysset_t
+//                                                  * identifying system calls traced on
+//
+//                                                  * entry.  If 0, then no entry syscalls
+//                                                  * are being traced. */
+//             uint32_t pr_sysexit_offset;         /* offset into pstatus file of sysset_t
+//                                                  * identifying system calls traced on
+//                                                  * exit.  If 0, then no exit syscalls
+//                                                  * are being traced. */
+//             uint64_t pr__pad[8];                /* reserved for future use */
+//             lwpstatus_t pr_lwp;                 /* "representative" thread status */
+//
+// From /usr/include/sys/procfs.h
+// typedef struct pr_sigset
+// {
+//    uint64_t ss_set[4];              /* signal set */
+// } pr_sigset_t;
+//
+// typedef struct pr_timestruc64
+// {
+//    int64_t  tv_sec;                 /* 64 bit time_t value      */
+//    int32_t  tv_nsec;                /* 32 bit suseconds_t value */
+//    uint32_t __pad;                  /* reserved for future use  */
+// } pr_timestruc64_t;
+//
+// typedef void * prptr64_t;
+//
+// The fields before the user cpu time (pr_utime) are:
+//	uint32_t pr_flag;		4				4
+//	uint32_t pr_flag2;		4				8
+//	uint32_t pr_flags;		4				12
+//	uint32_t pr_nlwp;		4				16
+//	char     pr_stat;		1				17
+//	char     pr_dmodel;		1				18
+//	char     pr__pad1[6];	6				24
+//	pr_sigset_t pr_sigpend;	(4 * 8) = 32	56
+//	prptr64_t pr_brkbase;	8				64
+//	uint64_t pr_brksize;	8				72
+//	prptr64_t pr_stkbase;	8				80
+//	uint64_t pr_stksize;	8				88
+//	pid64_t  pr_pid;		8				96
+//	pid64_t  pr_ppid;		8				104
+//	pid64_t  pr_pgid;		8				112
+//	pid64_t  pr_sid;		8				120
+//  total:                  120
+// followed by:
+//	struct pr_timestruc64_t pr_utime;   /* process user cpu time */
+
+// As explained above, we will skip 120 bytes into the status file to locate the user CPU time.
+const skip = 120
+
+func cpuTimeUser(pid int32) (float64, error) {
```

Comment:
A unit test might be overkill, but have you tested this yourself manually? Does it return the right value and does it result in the same metric in Datadog as on other OSs?

Reply:
Unit tests actually cover this already. I've run them on AIX.
They're unfortunately a little flaky since they're heuristic (see the note here: https://github.com/DataDog/datadog-agent/blob/master/pkg/trace/watchdog/info_test.go#L66-L69) so they fail from time to time. I would like to improve that as we move towards automating AIX testing.